services/secondary-storage/server/src/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java

```diff
@@ -942,6 +931,83 @@ protected Answer copyFromNfsToS3(CopyCommand cmd) {
         }
     }
 
+    protected boolean swiftUploadMetadataFile(SwiftTO swift, File srcFile, String containerName) throws IOException {
+
+
+        //create a template.properties for Swift with its correct unique name
```

Comment:
Thanks for the suggestion. I finally was able to get some time to work on this again. I'll send a new PR soon with your comments.